### ISSUE TITLE
Use mypy within docker

### ISSUE THREAD

**Issue by Euphorbium** [leinardi/mypy-pycharm#28]
### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [] I have verified that I am using the latest version of the plugin.

### Step 2: Describe your environment
 - Plugin version: `?`
 - PyCharm/IDEA version: PyCharm 2018.3.5 (Professional Edition)
Build #PY-183.5912.18, built on February 26, 2019
JRE: 1.8.0_152-release-1343-b28 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
 - Mypy version: `?`

#### Steps to reproduce:

  1. When using project interpreter in docker compose, it is impossible to specify mypy executable path, and I am constantly getting popups saying that mypy is missing. I have set up it to use my host system mypy, and even installed it in the docker interpreter, but the pop ups continue.

#### Expected Results:

it should be possible to use mypy within docker.

```Gradle
10:55	Packages installed successfully: Installed packages: 'mypy'

10:58	Mypy Plugin (Mypy missing)
			The project interpreter is missing Mypy, which is needed to properly check the imports.
			Install Mypy

```

**Comment by hmoravec** [comment]
I also agree this is an important feature as a typical user of Mypy is a professional developer who also uses Docker/Vagrant for development environment. It would be nice if it would be at least possible to use Mypy executable from environment that is different than the project environment. 

There is also the same problem for the Pylint plugin: https://github.com/leinardi/pylint-pycharm/issues/36

**Comment by aguestuser** [comment]
+1. our team would love to see this issue re-opened and worked on. we are *dying* to use this plugin, but the lack of support for dockerized virtualenvs is an absolute deal-breaker. but for this one defect, we would be using your plugin instead of the dropbox plugin, which we think is strictly worse. do maintainers have any plans to pick this up? would you accept contributions to help move it forward?

**Comment by gaqzi** [comment]
I got this working by making a little naïve wrapper script that invokes any command in a container for me (based on the name of the calling command).

Given the following script named `.execute-in-docker.sh`:

```bash
#!/bin/bash

: ${DEV_CONTAINER:=code_1}
SCRIPT_NAME=$(basename $0)

# Certain commands from IntelliJ issues absolute paths from the
# local filesystem, but the container has the root of the source
# tree set as the working directory. So just remove all the local
# paths and it'll all work out
PROJECT_ROOT_DIRECTORY="$(git rev-parse --show-toplevel)/"
function relativize() {
    echo $@ | sed "s#${PROJECT_ROOT_DIRECTORY}##"
}

function relativize-all-the-arguments() {
    local args=""

    for i in $@; do
        i=$(relativetize "$i")
        args="${args} ${i}"
    done

    echo ${args}
}

exec /usr/local/bin/docker exec ${DEV_CONTAINER} \
    /usr/bin/env $(relativetize "$SCRIPT_NAME") $(relativetize-all-the-arguments $@)

```

Then I went ahead and created symlinks for all the commands I want to run in my container:

```shell
$ cd bin
$ ln -s .execute-in-docker.sh mypy
$ ln -s .execute-in-docker.sh isort
$ cd ../
$ bin/mypy src/my_funky_tool.py
src/my_funky_tool.py:38:5: error: No return value expected
src/my_funky_tool.py:40:1: error: Argument 1 to "hello" has incompatible type "int"; expected "str"
$ bin/isort src/my_funky_tool.py
```

In case it helps someone else that is looking to make this plugin work with Docker :)

**Comment by rommguy** [comment]
@leinardi Thank you very much for this plugin, it is super helpful
Perhaps it is possible to add interpreter settings to the plugin settings?
This would solve the problem of running in a docker, since I want PyCharm to use the remote interpreter but the plugin to use the local interpreter
I'd be happy to submit a PR if you can give me some guidance

**Comment by jaredtkatz** [comment]
Any progress here? Kinda frustrating :/
